### DISCUSSION TITLE
Why do I get this error?

### DISCUSSION THREAD

**Discussion by marvic2409** [vercel/vercel#5186] [Help] [answered]
I get this error when trying to deploy my python flask project:

`

  | Failed to run "pip install --disable-pip-version-check --target . --upgrade -r /vercel/50a63a87/requirements.txt"...
-- | --
23:22:23.303 | Error: Command failed: pip3 install --disable-pip-version-check --target . --upgrade -r /vercel/50a63a87/requirements.txt
23:22:23.303 | ERROR: Command errored out with exit status 1:
23:22:23.303 | command: /usr/local/bin/python3.6 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-sgwz2a81/psycopg2/setup.py'"'"'; __file__='"'"'/tmp/pip-install-sgwz2a81/psycopg2/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-a3sfeilr
23:22:23.303 | cwd: /tmp/pip-install-sgwz2a81/psycopg2/
23:22:23.303 | Complete output (23 lines):
23:22:23.303 | running egg_info
23:22:23.303 | creating /tmp/pip-pip-egg-info-a3sfeilr/psycopg2.egg-info
23:22:23.304 | writing /tmp/pip-pip-egg-info-a3sfeilr/psycopg2.egg-info/PKG-INFO
23:22:23.304 | writing dependency_links to /tmp/pip-pip-egg-info-a3sfeilr/psycopg2.egg-info/dependency_links.txt
23:22:23.304 | writing top-level names to /tmp/pip-pip-egg-info-a3sfeilr/psycopg2.egg-info/top_level.txt
23:22:23.304 | writing manifest file '/tmp/pip-pip-egg-info-a3sfeilr/psycopg2.egg-info/SOURCES.txt'
23:22:23.304 |  
23:22:23.304 | Error: pg_config executable not found.
23:22:23.304 |  
23:22:23.304 | pg_config is required to build psycopg2 from source.  Please add the directory
23:22:23.304 | containing pg_config to the $PATH or specify the full executable path with the
23:22:23.304 | option:
23:22:23.304 |  
23:22:23.304 | python setup.py build_ext --pg-config /path/to/pg_config build ...
23:22:23.304 |  
23:22:23.304 | or with the pg_config option in 'setup.cfg'.
23:22:23.304 |  
23:22:23.304 | If you prefer to avoid building psycopg2 from source, please install the PyPI
23:22:23.304 | 'psycopg2-binary' package instead.
23:22:23.304 |  
23:22:23.304 | For further information please check the 'doc/src/install.rst' file (also at
23:22:23.304 | <https://www.psycopg.org/docs/install.html>).
23:22:23.304 |  
23:22:23.304 | ----------------------------------------
23:22:23.304 | ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
23:22:23.304 | Collecting aniso8601==8.0.0
23:22:23.304 | Downloading aniso8601-8.0.0-py2.py3-none-any.whl (43 kB)
23:22:23.304 | Collecting attrs==20.2.0
23:22:23.304 | Downloading attrs-20.2.0-py2.py3-none-any.whl (48 kB)
23:22:23.304 | Collecting click==7.1.2
23:22:23.304 | Downloading click-7.1.2-py2.py3-none-any.whl (82 kB)
23:22:23.304 | Collecting Flask==1.1.2
23:22:23.304 | Downloading Flask-1.1.2-py2.py3-none-any.whl (94 kB)
23:22:23.304 | Collecting flask-restplus==0.13.0
23:22:23.305 | Downloading flask_restplus-0.13.0-py2.py3-none-any.whl (2.5 MB)
23:22:23.305 | Collecting Flask-SQLAlchemy==2.4.4
23:22:23.305 | Downloading Flask_SQLAlchemy-2.4.4-py2.py3-none-any.whl (17 kB)
23:22:23.305 | Collecting itsdangerous==1.1.0
23:22:23.305 | Downloading itsdangerous-1.1.0-py2.py3-none-any.whl (16 kB)
23:22:23.305 | Collecting Jinja2==2.11.2
23:22:23.305 | Downloading Jinja2-2.11.2-py2.py3-none-any.whl (125 kB)
23:22:23.305 | Collecting jsonschema==3.2.0
23:22:23.305 | Downloading jsonschema-3.2.0-py2.py3-none-any.whl (56 kB)
23:22:23.305 | Collecting MarkupSafe==1.1.1
23:22:23.305 | Downloading MarkupSafe-1.1.1-cp36-cp36m-manylinux1_x86_64.whl (27 kB)
23:22:23.305 | Collecting psycopg2==2.8.6
23:22:23.305 | Downloading psycopg2-2.8.6.tar.gz (383 kB)
23:22:23.305 | at makeError (/vercel/cf4b7156fd2ef8b5/.build-utils/.builder/node_modules/@vercel/python/dist/index.js:2189:9)
23:22:23.305 | at /vercel/cf4b7156fd2ef8b5/.build-utils/.builder/node_modules/@vercel/python/dist/index.js:2293:16
23:22:23.305 | at processTicksAndRejections (internal/process/task_queues.js:97:5)
23:22:23.305 | at async pipInstall (/vercel/cf4b7156fd2ef8b5/.build-utils/.builder/node_modules/@vercel/python/dist/index.js:3748:9)
23:22:23.305 | at async Object.installRequirementsFile (/vercel/cf4b7156fd2ef8b5/.build-utils/.builder/node_modules/@vercel/python/dist/index.js:3778:5)
23:22:23.305 | at async Object.module.exports.__webpack_modules__.855.exports.build (/vercel/cf4b7156fd2ef8b5/.build-utils/.builder/node_modules/@vercel/python/dist/index.js:3644:9)
23:22:23.305 | at async buildStep (/var/task/sandbox-worker.js:21075:20)
23:22:23.305 | at async mainSub (/var/task/sandbox-worker.js:20796:9)
23:22:23.305 | at async main (/var/task/sandbox-worker.js:20658:5) {
23:22:23.305 | code: 1,
23:22:23.305 | stdout: 'Collecting aniso8601==8.0.0\n' +
23:22:23.305 | '  Downloading aniso8601-8.0.0-py2.py3-none-any.whl (43 kB)\n' +
23:22:23.305 | 'Collecting attrs==20.2.0\n' +
23:22:23.305 | '  Downloading attrs-20.2.0-py2.py3-none-any.whl (48 kB)\n' +
23:22:23.305 | 'Collecting click==7.1.2\n' +
23:22:23.305 | '  Downloading click-7.1.2-py2.py3-none-any.whl (82 kB)\n' +
23:22:23.305 | 'Collecting Flask==1.1.2\n' +
23:22:23.305 | '  Downloading Flask-1.1.2-py2.py3-none-any.whl (94 kB)\n' +
23:22:23.306 | 'Collecting flask-restplus==0.13.0\n' +
23:22:23.306 | '  Downloading flask_restplus-0.13.0-py2.py3-none-any.whl (2.5 MB)\n' +
23:22:23.306 | 'Collecting Flask-SQLAlchemy==2.4.4\n' +
23:22:23.306 | '  Downloading Flask_SQLAlchemy-2.4.4-py2.py3-none-any.whl (17 kB)\n' +
23:22:23.306 | 'Collecting itsdangerous==1.1.0\n' +
23:22:23.306 | '  Downloading itsdangerous-1.1.0-py2.py3-none-any.whl (16 kB)\n' +
23:22:23.306 | 'Collecting Jinja2==2.11.2\n' +
23:22:23.306 | '  Downloading Jinja2-2.11.2-py2.py3-none-any.whl (125 kB)\n' +
23:22:23.306 | 'Collecting jsonschema==3.2.0\n' +
23:22:23.306 | '  Downloading jsonschema-3.2.0-py2.py3-none-any.whl (56 kB)\n' +
23:22:23.306 | 'Collecting MarkupSafe==1.1.1\n' +
23:22:23.306 | '  Downloading MarkupSafe-1.1.1-cp36-cp36m-manylinux1_x86_64.whl (27 kB)\n' +
23:22:23.306 | 'Collecting psycopg2==2.8.6\n' +
23:22:23.306 | '  Downloading psycopg2-2.8.6.tar.gz (383 kB)\n',
23:22:23.306 | stderr: '    ERROR: Command errored out with exit status 1:\n' +
23:22:23.306 | `     command: /usr/local/bin/python3.6 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-sgwz2a81/psycopg2/setup.py'"'"'; __file__='"'"'/tmp/pip-install-sgwz2a81/psycopg2/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\\r\\n'"'"', '"'"'\\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-a3sfeilr\n` +
23:22:23.306 | '         cwd: /tmp/pip-install-sgwz2a81/psycopg2/\n' +
23:22:23.306 | '    Complete output (23 lines):\n' +
23:22:23.306 | '    running egg_info\n' +
23:22:23.306 | '    creating /tmp/pip-pip-egg-info-a3sfeilr/psycopg2.egg-info\n' +
23:22:23.306 | '    writing /tmp/pip-pip-egg-info-a3sfeilr/psycopg2.egg-info/PKG-INFO\n' +
23:22:23.306 | '    writing dependency_links to /tmp/pip-pip-egg-info-a3sfeilr/psycopg2.egg-info/dependency_links.txt\n' +
23:22:23.306 | '    writing top-level names to /tmp/pip-pip-egg-info-a3sfeilr/psycopg2.egg-info/top_level.txt\n' +
23:22:23.306 | "    writing manifest file '/tmp/pip-pip-egg-info-a3sfeilr/psycopg2.egg-info/SOURCES.txt'\n" +
23:22:23.306 | '    \n' +
23:22:23.306 | '    Error: pg_config executable not found.\n' +

`

**Comment by oyedejioyewole** [comment]
Are you trying to use PostgreSQL for django db

**Comment by oyedejioyewole** [comment]
Because Vercel doesn't provide already compiled binaries for `psycopg2` library

Meaning you have to clone the this repository <https://github.com/jkehler/awslamda-psycopg2> and copy the folder `psycopg2-3.6` or folder name like that

Paste folder to the `root` where your **vercel.json** or **now.json** file lies

Then update requirements.txt file like this `psycopg2==2.7.7`

Afterwards then run the command `vercel --prod` or `vercel`

It worked for me, so you can try it

**Reply by oyedejioyewole** [reply]
Forgot to say this worked for me because I was using `Django` not `Flask`

**Reply by ChikangaTakudzwa** [reply]
i am having problems deploying django on vercel, any resources or tips would help thanks

**Comment by ChengxiX** [comment, marked as answer]
now [this repository](https://github.com/jkehler/awslamda-psycopg2) become invalid. but can simply change requirements.txt to 
`psycopg2-binary~=2.9.3`

**Reply by oyedejioyewole** [reply]
thanks for the update, 👍🏼

**Reply by ChikangaTakudzwa** [reply]
was also getting the same error thanks

**Reply by punyapatkha** [reply]
this fix my problem when deploy django on vercel. thank you so much

**Reply by SandeepK1729** [reply]
> psycopg2-binary~=2.9.3

Thanks for update

**Comment by qestebane** [comment]
Thanks for suggesting it help me. This is the repository update link from repository: https://github.com/jkehler/awslambda-psycopg2
